nixos/modules/services/networking/unbound.nix

```
@@ -21,7 +21,17 @@ let
                                 ))
     else throw (traceSeq v "services.unbound.settings: unexpected type");
 
-  confFile = pkgs.writeText "unbound.conf" (concatStringsSep "\n" ((mapAttrsToList (toConf "") cfg.settings) ++ [""]));
+  confNoServer = concatStringsSep "\n" ((mapAttrsToList (toConf "") (builtins.removeAttrs cfg.settings [ "server" ])) ++ [""]);
+  confServer = concatStringsSep "\n" (mapAttrsToList (toConf "  ") (builtins.removeAttrs cfg.settings.server [ "define-tag" ]));
+
+  confFile = pkgs.writeText "unbound.conf" ''
+    server:
+    ${optionalString (cfg.settings.server.define-tag != []) ''
+      define-tag: "${concatStringsSep " " cfg.settings.server.define-tag}"
```

wondering whether wrapping in `"` is good here as it breaks with how the rest of the module treats options (and might break define-tag settings that worked previously). treating it differently does seem more friendly, but we should probably document that with an explicit option in the manual and/or forbid `"` in tags

---

According to the man page:

> Enclose the list between quotes ("") and put spaces between tags.

What do you propose we do otherwise?

---

either:
```suggestion
      define-tag: ${concatStringsSep " " cfg.settings.server.define-tag}
```

because `define-tag` isn't documented as having this special automatic behaviour, or adding a new option to `settings` to document it instead:

```
server.define-tag = mkOption {
  type = with types; listOf strMatching "[^"\n\r\t ]+" // {
    description = "single-line string without `"`";
  };
  description = "List of unbound tags. Surrounding `"` will be added automatically."
  default = [ ];
  example = [ "local" ];
};
```

both are fine, we just think that it's a possible source of avoidable confusion to have one option treat it's values differently than all others without stating that it does

---

I'll change it to use the `toOption` function that is already defined. You will then be able to define this option with:
```
server.define-tag = ''"local test"'';
```